kpatch-build/create-diff-object.c

```diff
@@ -975,15 +975,23 @@ static struct symbol *kpatch_find_static_twin(struct section *sec,
 
 	/* find the patched object's corresponding variable */
 	list_for_each_entry(rela, &sec->twin->relas, list) {
+		struct symbol *patched_sym;
 
 		rela_toc = toc_rela(rela);
 		if (!rela_toc)
 			continue; /* skip toc constants */
 
-		if (rela_toc->sym->twin)
+		patched_sym = rela_toc->sym;
+
+		if (patched_sym->twin)
+			continue;
+
+		if (sym->type != patched_sym->type ||
+		    (sym->type == STT_OBJECT &&
+		     sym->sym.st_size != patched_sym->sym.st_size))
 			continue;
 
-		if (kpatch_mangled_strcmp(rela_toc->sym->name, sym->name))
+		if (kpatch_mangled_strcmp(patched_sym->name, sym->name))
 			continue;
 
 		return rela_toc->sym;
```

Comment:
Might as well return `patched_sym` here for readability.

Reply:
You're right, I forgot to replace this reference.
Thanks for spotting it.

I'll update this along with the submodule update once dynup/kpatch-unit-test-objs#20 is merged.